src/main/java/Duke.java

```diff
@@ -1,10 +1,276 @@
+    }
+}
+
+class DeadlineEmptyBodyException extends EmptyBodyException {
+    DeadlineEmptyBodyException() {}
+    @Override
+    public String toString() {
+        return "☹ OOPS!!! The description of a deadline cannot be empty.";
+    }
+}
+
+class DeleteEmptyBodyException extends EmptyBodyException {
+    DeleteEmptyBodyException() {}
+    @Override
+    public String toString() {
+        return "☹ OOPS!!! Empty deletion is invalid.";
+    }
+}
+
+class DoneEmptyBodyException extends EmptyBodyException {
+    DoneEmptyBodyException() {}
+    @Override
+    public String toString() {
+        return "☹ OOPS!!! Empty done is invalid.";
+    }
+}
+
+class InvalidDeletionException extends Exception {
+    InvalidDeletionException() {}
+    @Override
+    public String toString() {
+        return "☹ OOPS!!! Deletion index is invalid.";
+    }
+}
+
+class InvalidDoneException extends Exception {
+    InvalidDoneException() {}
+    @Override
+    public String toString() {
+        return "☹ OOPS!!! Done index is invalid.";
+    }
+}
+
+class Todo extends Task {
+    Todo(String message) {
+        super(message);
+    }
+
+    @Override
+    public String getTypeLetter() {
+        // dummy value
+        return "[T]";
+    }
+}
+
+class Event extends Task {
+    Event(String message) {
+        super(message);
+    }
+
+    @Override
+    public String getTypeLetter() {
+        // dummy value
+        return "[E]";
+    }
+}
+
+class Deadline extends Task {
+    Deadline(String message) {
+        super(message);
+    }
+
+    @Override
+    public String getTypeLetter() {
+        // dummy value
+        return "[D]";
+    }
+}
+
+class Convert {
+    static String at(String s) {
+        String first = s.split("/at")[0];
+        String second = s.split("/at")[1];
+        return first + "(at:" + second + ")";
+    }
+
+    static String by(String s) {
+        String first = s.split("/by")[0];
+        String second = s.split("/by")[1];
+        return first + "(by:" + second + ")";
+    }
+}
+
 public class Duke {
+    // constant SPACE and LINE for format purposes
+    public static String SPACE = "     ";
+    public static String LINE = "____________________________________________________________\n";
+    // add outer frame lines
+    public static String format(String input) {
+        return SPACE + LINE +
+               SPACE + " " + input + "\n" +
+               SPACE + LINE;
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        // greeting and exit messages strings
+        // list and mark strings
+        final String messageHello = "Hello! I'm Duke\n" + SPACE + " " + "What can I do for you?";
```

Comment:
Should these constants be named in uppercase and separate with underscores between words?